kernel/kernel-auth-service/src/main/java/io/mosip/kernel/auth/service/impl/ProxyOTPServiceImpl.java

```diff
@@ -0,0 +1,705 @@
+
+			if (ex.getRawStatusCode() == 401) {
+				if (!validationErrorsList.isEmpty()) {
+					throw new AuthNException(validationErrorsList);
+				} else {
+					throw new AuthManagerException(AuthErrorCode.CLIENT_ERROR.getErrorCode(),
+							AuthErrorCode.CLIENT_ERROR.getErrorMessage(), ex);
+				}
+			}
+			if (ex.getRawStatusCode() == 403) {
+				if (!validationErrorsList.isEmpty()) {
+					throw new AuthZException(validationErrorsList);
+				} else {
+					throw new AuthManagerException(AuthErrorCode.CLIENT_ERROR.getErrorCode(), ex.getMessage(), ex);
+				}
+			}
+			if (!validationErrorsList.isEmpty()) {
+				throw new AuthManagerServiceException(validationErrorsList);
+			} else {
+				throw new AuthManagerException(AuthErrorCode.CLIENT_ERROR.getErrorCode(), ex.getMessage(), ex);
+			}
+		}
+		return otpEmailSendResponseDto;
+	}
+
+	private AccessTokenResponse getUserAccessToken(String username, String realm) {
+		HttpHeaders headers = new HttpHeaders();
+		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+		MultiValueMap<String, String> tokenRequestBody = null;
+		Map<String, String> pathParams = new HashMap<>();
+		pathParams.put(AuthConstant.REALM_ID, realm);
+		UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromUriString(keycloakOpenIdUrl + "/token");
+		tokenRequestBody = getAdminValueMap(username, realm);
+		HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(tokenRequestBody, headers);
+		ResponseEntity<AccessTokenResponse> response = restTemplate.postForEntity(
+				uriComponentsBuilder.buildAndExpand(pathParams).toUriString(), request, AccessTokenResponse.class);
+		return response.getBody();
+	}
+
+	private AccessTokenResponse getAuthAccessToken(String clientID, String clientSecret, String realmId) {
+		HttpHeaders headers = new HttpHeaders();
+		if (realmId.equalsIgnoreCase(preregRealmId)) {
+			clientID = preregClientId;
+			clientSecret = preregSecretKey;
+		}
+		headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+		MultiValueMap<String, String> tokenRequestBody = null;
+		Map<String, String> pathParams = new HashMap<>();
+		pathParams.put(AuthConstant.REALM_ID, realmId);
+		UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromUriString(keycloakOpenIdUrl + "/token");
+		tokenRequestBody = getClientValueMap(clientID, clientSecret);
+		HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(tokenRequestBody, headers);
+		ResponseEntity<AccessTokenResponse> response = restTemplate.postForEntity(
+				uriComponentsBuilder.buildAndExpand(pathParams).toUriString(), request, AccessTokenResponse.class);
+		return response.getBody();
+	}
+
+	private MultiValueMap<String, String> getAdminValueMap(String username, String realm) {
+		MultiValueMap<String, String> map = new LinkedMultiValueMap<>();
+		if (realm.equalsIgnoreCase(preregRealmId)) {
```

Comment:
Why should we differentiate between prereg and admin portal?

Reply:
Will change this to unsupported Operation